pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -569,20 +570,38 @@ func (c *Controller) getProxyServiceInstancesByEndpoint(endpoints v1.Endpoints,
 					continue
 				}
 
-				// There is only one IP for kube registry
 				proxyIP := proxy.IPAddresses[0]
 
 				if hasProxyIP(ss.Addresses, proxyIP) {
 					out = append(out, c.getEndpoints(proxyIP, port.Port, svcPort, svc))
 				}
 
 				if hasProxyIP(ss.NotReadyAddresses, proxyIP) {
-					nrEP := c.getEndpoints(proxyIP, port.Port, svcPort, svc)
-					out = append(out, nrEP)
+					out = append(out, c.getEndpoints(proxyIP, port.Port, svcPort, svc))
 					if c.Env != nil {
 						c.Env.PushContext.Add(model.ProxyStatusEndpointNotReady, proxy.ID, proxy, "")
 					}
 				}
+
+				// copy from the first element in the array, then change the ip and network when there are multiple ips
+				for index, ip := range proxy.IPAddresses {
+					if index != 0 && len(out) > 0 {
+						deepCopy := deepCopyServiceInstance(out[0])
+						deepCopy.Endpoint.Address = ip
```

Comment:
This does not make sense to me, why are we modifying `out[0]`? If we have three Ip address then we modify it twice and just override it each time? And do a full deepcopy both times?

Reply:
I want to generate a ServiceInstance for each IP, In the case of multiple IPS, a deep copy is made based on the first record, and then override the IP and Network properties.

Reply:
Already modified